backend/src/utils/utils.ts

```diff
@@ -16,3 +21,40 @@ const runShellCommand = async (command: string): Promise<string> => {
 
 export const pipe = <T>(...fns: Array<(a: T) => T>) => (x: T): T =>
   fns.reduce((value, func) => func(value), x)
+
+export const getRepositoryFromFilePath = (file: File): string => {
+  return file.name.split('/').slice(0, 2).join('/')
+}
+
+export const getFileNameFromFilePath = (
+  file: File,
+  repositoryName: string
+): string => {
+  return file.name.replace(`${repositoryName}/`, '') || file.name
+}
+
+export const writeToFile = (file: File): void => {
+  writeFileSync(`./repositories/${file.name}`, file.content)
+}
+
+export const makeCommitMessage = (
+  rawCommitMessage: string,
+  username: string,
+  realFilename: string
+): string => {
+  return rawCommitMessage
+    ? sanitizeCommitMessage(rawCommitMessage)
+    : `User ${username} modified file ${realFilename}`
+}
+
+export const getRepoLocationFromUrlString = (urlString: string): string => {
+  const url = new URL(urlString)
+  const repositoryName = url.pathname
+  const repoLocation = `./repositories${repositoryName}`
+  return repoLocation
```

Comment:
You can use getRepoLocationFromRepoName() function here to reduce repetitiveness

Reply:
Not the same end result imo. One extra slash.

Reply:
You are right. Didn't see the extra slash